forte_wrapper/twitter/twittersearch_processor.py

```diff
@@ -0,0 +1,133 @@
+
+        Returns:
+            dictionary with the default config for this processor.
+        Following are the keys for this dictionary:
+        - `"credential_file"`: defines the path of credential file needed for
+            Twitter API usage.
+        - `"num_tweets_returned"`: defines the number of tweets returned by
+            processor.
+        - `"lang"`: language, restricts tweets to the given language,
+            default is 'en'.
+        - `"date_since"`: restricts tweets created after the given date.
+        - `"result_type"`: defines what type of search results to receive.
+            The default is “recent.” Valid values include:
+            mixed : include both popular and real time results in the response
+            recent : return only the most recent results in the response
+            popular : return only the most popular results in the response
+        - `"query_pack_name"`: The query pack's name, default is "query".
+        - `"response_pack_name_prefix"`: the pack name prefix to be used
+                in response datapacks.
+        """
+        # pylint: enable=line-too-long
+        config = super().default_configs()
+        config.update({
+            "credential_file": "",
+            "num_tweets_returned": 5,
+            "lang": "en",
+            "date_since": "2020-01-01",
+            "result_type": 'recent',
+            "query_pack_name": "query",
+            "response_pack_name_prefix": "passage"
+        })
+        return config
+
+    def _process(self, input_pack: MultiPack):
+        r"""Search using Twitter API to fetch tweets for a query.
+        This query should be contained in the input multipack with name
+        `self.config.query_pack_name`.
+        Each result is added as a new data pack, and a
+        `ft.onto.base_ontology.Document` annotation is used to cover the whole
+        document.
+        Args:
+             input_pack: A multipack containing query as a pack.
+        """
+        query_pack = input_pack.get_pack(self.configs.query_pack_name)
+
+        query = query_pack.text
+        tweets = self._query_tweets(query)
+
+        for idx, tweet in enumerate(tweets):
+            try:
+                text = tweet.retweeted_status.full_text
+
+            except AttributeError:  # Not a Retweet
+                text = tweet.full_text
+
+            pack: DataPack = input_pack.add_pack(
+                f"{self.configs.response_pack_name_prefix}_{idx}"
+            )
+            pack.pack_name = f"{self.configs.response_pack_name_prefix}_{idx}"
+
+            pack.set_text(text)
+
+            Document(pack=pack, begin=0, end=len(text))
+
+    def _query_tweets(self, query: str):
+        """
+        This function searches tweets using Twitter API
+        Args:
+            query: user's input query for twitter API search
+        Returns: List of tweets
```

Comment:
In general, there should be new lines separating the main docstring text, the `Args` block, and the `Returns` block.

Something like here: https://github.com/asyml/forte/blob/master/forte/data/data_pack.py#L597